CONFIGURATION.md

```diff
@@ -0,0 +1,201 @@
+```
+
+## Proposal: Configuration objects can be referenced using YAML references
+
+In order to re-use a complex configuration object (like an origin, a mount mapping, or even a strain), use YAML references (`*publish`) or YAML extensions (`<< *basestrain`). This allows us to rely on the behavior of the YAML parser instead of using home-grown inheritance and referencing logic that is likely less robust and certainly less well documented.
+
+A strain re-using the `*basestrain` above would look like this:
+
+```yaml
+  client:
+    <<: *basestrain
+    condition: req.http.host == "client.project-helix.io"
+    content:
+      repo: helix-cli
+      ref: master
+      owner: adobe
+    directoryIndex: readme.html
+```
+
+A strain referencing the `*publish` origin would look like this:
+```yaml
+  proxy-detailed:
+    sticky: true
+    origin: *publish
+```
+
+## Proposal: The `default` strain serves all traffic in production when no other strain is selected, therefore a `default` strain is mandatory
+
+We need one strain that is serving traffic that no other strain claims. This could be one by:
+
+1. requiring the existence of a `default` strain
+2. requiring the existence of one and only one strain that has the `default: true` property set
+
+Option (1) is easier to validate than option (2) and less error prone (every YAML editor will tell you if you have two keys called `default`), so I'd opt for (1)
+
+`hlx publish` should fail when no `default` strain exists.
+`hlx up` can continue without a `default` strain, but should issue a warning
+
+## Proposal: There is no implicit inheritance from `default`
+
+All inheritance becomes explicit, using the mechanism laid out in 
+https://github.com/adobe/helix-shared/issues/11#issuecomment-443122026. This means the `default` strain is special only in regards to serving traffic, but does not alter the behavior of any other strain.
+
+## Proposal: Strains can have a `mount` configuration that maps URL paths to content paths
+
+Right now, strains can have only one `content` node. With support for `mount`, a strain can have either a `content` or a `mount` property. Each `mount` object is a map between the URL path (key) and a `content` property (either git URL or object).
+
+```yaml
+strains:
+  default:
+    mount:
+      /: https://github.com/adobe/project-helix.io.git
+      /pipeline: https://github.com/adobe/helix-pipeline.git
+      /cli: https://github.com/adobe/helix-cli.git
+```
+
+## Proposal: `mount` points can take Git URLs, `content` objects/references, Backend URLs, `origin` objects/references
+
+That way you can have a consistent traffic mapping without having to switch strains.
+
+## Proposal: `hlx deploy` lists all strains that are affected by the deployment and suggests the creation of a new strain if none are affected
+
+In order to increase the visibility of changes happening during deployment, `hlx deploy` will list all strain names that will be affected by the deployment.
```

Comment:
who is this list determined? what selects a strain based on the current git-repo? note, that we can also have pure-config git-repos.

Reply:
If `hlx deploy` does not have access to `helix-config.yaml` we won't be able to build that list (this would be the case for a separate config repo). I would assume that in this case you already know what you are doing and don't need hand-holding in updating your config.

Reply:
How is the list generated? how is it different from just 'all' strains? how does a strain select the actions?